### ISSUE TITLE
Better name for Tuner methods

### ISSUE THREAD

**Issue by awaelchli** [Lightning-AI/lightning#16529]
### Description & Motivation

The PR #16462 introduced a breaking change with Lightning 2.0 that moves the `auto_lr_find` and `auto_scale_batch_size` features out of the Trainer:

```py
trainer = Trainer()
tuner = Tuner(trainer)
tuner.lr_find(...)
tuner.scale_batch_size(...)
```

There is now an opportunity to also rename the methods, if we feel like there are better, more intuitive terms out there.

### Pitch

Learning Rate tuner:

```py
# current
tuner.lr_find()

# A
tuner.find_lr()

# B
tuner.find_learning_rate()

# C
(your suggestion in the comment)
```

Batch Size tuner:

```py
# current
tuner.scale_batch_size()

# A (@Borda)
tuner.maximize_batch_size()

# B
tuner.find_batch_size()

# C
(your suggestion in the comment)
```

### Alternatives

Keep as is :) 

cc @borda @tchaton @justusschock @awaelchli

**Comment by Borda** [comment]
Maybe sometime in the future...
